### PR TITLE
Update and add dev guide content

### PR DESCRIPTION
**Pull request by rhornung67**
# Summary

- This PR adds content to the RAJA Dev Guide about CI test organization and maintenance tasks.

The generated documentation can be viewed here: https://raja.readthedocs.io/en/task-ci-docs/sphinx/dev_guide/index.html

### CODE REVIEW

**Comment by kab163** [comment]
Is there a link to the readthedocs that this generates?

**Review by kab163 — COMMENTED** [review]
Should there be any mentioning of how to create a new spec job? (Or updating a job if updates are done to a LC machine to where a spec no longer works?) That may require adding certain flags in the `scripts/spack_packages/raja/package.py` file or adding other parameters to the spec like `cxxflags=...`

**Comment by rhornung67** [comment]
> Should there be any mentioning of how to create a new spec job? (Or updating a job if updates are done to a LC machine to where a spec no longer works?) That may require adding certain flags in the `scripts/spack_packages/raja/package.py` file or adding other parameters to the spec like `cxxflags=...`

Yes. I am planning to add a section on CI maintenance tasks that will cover those sorts of things.

File: docs/sphinx/dev_guide/build_configurations.rst
Comment (rhornung67) [reply docs/sphinx/dev_guide/build_configurations.rst:117]: that section is old and needs to be updated...

File: docs/sphinx/dev_guide/build_configurations.rst
Comment (rhornung67) [reply docs/sphinx/dev_guide/build_configurations.rst:142]: ditto.

File: docs/sphinx/dev_guide/ci.rst
Comment (rhornung67) [reply docs/sphinx/dev_guide/ci.rst:98]: That's the figure caption.

File: docs/sphinx/dev_guide/ci.rst
Comment (rhornung67) [reply docs/sphinx/dev_guide/ci.rst:136]: Dunno. I must have been sleepy.... I will fix.

File: docs/sphinx/dev_guide/ci.rst
Comment (kab163) [reply docs/sphinx/dev_guide/ci.rst:98]: Perhaps "This figure shows the directories and files..."

**Comment by kab163** [comment]
I will think about this over the weekend. If I think about other things that could be added here, I'll let you know. So far, this looks like very helpful, much needed documentation. Thanks Rich!

File: docs/sphinx/dev_guide/build_configurations.rst
Comment (rhornung67) [reply docs/sphinx/dev_guide/build_configurations.rst:107]: fixed.

File: docs/sphinx/dev_guide/build_configurations.rst
Comment (rhornung67) [reply docs/sphinx/dev_guide/build_configurations.rst:117]: fixed.

File: docs/sphinx/dev_guide/build_configurations.rst
Comment (rhornung67) [reply docs/sphinx/dev_guide/build_configurations.rst:142]: fixed.

File: docs/sphinx/dev_guide/build_configurations.rst
Comment (rhornung67) [reply docs/sphinx/dev_guide/build_configurations.rst:173]: fixed.

File: docs/sphinx/dev_guide/ci.rst
Comment (rhornung67) [reply docs/sphinx/dev_guide/ci.rst:66]: I address your first concern. 

I'm not clear on what to say about mirroring. If you have any thoughts to share about what someone unfamiliar with it should know, please let me know, or feel free to edit the docs.

File: docs/sphinx/dev_guide/ci.rst
Comment (rhornung67) [reply docs/sphinx/dev_guide/ci.rst:98]: fixed.

File: docs/sphinx/dev_guide/ci.rst
Comment (rhornung67) [reply docs/sphinx/dev_guide/ci.rst:135]: done.

File: docs/sphinx/dev_guide/ci.rst
Comment (rhornung67) [reply docs/sphinx/dev_guide/ci.rst:138]: fixed.

**Comment by rhornung67** [comment]
@kab163 I addressed all your comments, except for adding more info about Gitlab mirroring. I also, added links to files in the repo that may help folks navigate more easily.

**Comment by rhornung67** [comment]
@mdavis36 if you have some time, please take a crack at filling in some content in the Azure CI maintenance tasks section in the dev guide, specifically, w.r.t. to the docker images we use for the builds.

**Comment by MrBurmark** [comment]
This looks good. I learning a lot about the CI config and spack stuff. I also liked the section about writing a test.

**Comment by rhornung67** [comment]
Thanks @MrBurmark. I will take another clean up pass and add a few missing things that I recently learned. Then, I will remove draft status and it should be good to go.

**Review by artv3 — DISMISSED** [review]

**Review by kab163 — DISMISSED** [review]

**Comment by rhornung67** [comment]
Thanks for the approvals. Anyone else want to review and approve before I merge?

**Comment by rchen20** [comment]
Taking a quick look now.

File: docs/sphinx/dev_guide/ci_tasks.rst
Comment (rchen20) [inline docs/sphinx/dev_guide/ci_tasks.rst:190]:
```suggestion
    **Sycl** requires us to run setupvars.sh::
```

File: docs/sphinx/dev_guide/release_process.rst
Comment (rchen20) [inline docs/sphinx/dev_guide/release_process.rst:31]: Just to clarify, we create a release candidate branch from `develop`, but merge it into `main`?

File: docs/sphinx/dev_guide/release_process.rst
Comment (rchen20) [inline docs/sphinx/dev_guide/release_process.rst:164]: RAJAProxies mentioned twice, although both statements are slightly different.

File: docs/sphinx/dev_guide/ci_tasks.rst
Comment (rhornung67) [reply docs/sphinx/dev_guide/ci_tasks.rst:190]: fixed.

File: docs/sphinx/dev_guide/release_process.rst
Comment (rchen20) [reply docs/sphinx/dev_guide/release_process.rst:31]: Thanks, so `main` only has updates up until the latest release.

File: docs/sphinx/dev_guide/release_process.rst
Comment (rchen20) [inline docs/sphinx/dev_guide/release_process.rst:161]:
```suggestion
    updating the submodules to the new RAJA Portability Suite project 
```

**Review by rchen20 — DISMISSED** [review]

File: docs/sphinx/dev_guide/release_process.rst
Comment (rhornung67) [reply docs/sphinx/dev_guide/release_process.rst:31]: that is correct.

**Comment by rhornung67** [comment]
Folks, I took a last pass and fixed several typos, etc. unfortunately, this dismissed your approvals. Please re-approve and I'll merge this when it comes back green.

**Review by rchen20 — APPROVED** [review]

**Pull request merged**
